Document/0x05b-Basic-Security_Testing.md

```diff
@@ -1244,13 +1244,43 @@ For enterprise tools, see the section "Static Source Code Analysis" in the chapt
 
 #### Dynamic Analysis
 
--- ToDo: <https://github.com/OWASP/owasp-mstg/issues/1240>
+Dynamic Analysis tests the release version of an application, in a similar fashion to black box application testing. The way the code works should be inferred and looked at from a higher level, unlike static analysis, which takes the application's code and tries to analyze how the application will behave. This allows the tester to properly identify:
+
+- Business logic flaws which extend over multiple services
+- Vulnerabilities in live environments
+- Weak input validation and bad input/output encoding as they are processed through one or multiple services
+
+ whereas static analysis would identify:
+
+- Code bad practices
+- Regex usage issues
+- Hidden code layers
+
+Analysis can be assisted by automated tools, such as [MobSF](https://github.com/MobSF/Mobile-Security-Framework-MobSF/), while assessing an application. An application can be assessed by side-loading it, re-packaging it, or by simply attacking the installed version.
 
 ##### Using Non-Rooted Devices
 
-##### Method Tracing
+Non-rooted devices provide the tester with two benefits:
+
+- Replicate the same environment of the installed application
+- Use any device without the need of having it rooted
+
+In order to dynamically analyze the application, the tester could rely on [Objection](https://github.com/sensepost/objection) which is powered by Frida. This tool works by [patching](https://github.com/sensepost/objection/wiki/Patching-Android-Applications#patching---patching-an-apk) the target application with the [frida-gadget](https://www.frida.re/docs/gadget/) library. Objection communicates using a Python API with the mobile phone through the installed Gadget application.
```

Comment:
Please rephrase:

In order to dynamically analyze the application, you can rely on [objection](https://github.com/sensepost/objection) which is leveraging Frida. One way to use objection is to [patch the APK](https://github.com/sensepost/objection/wiki/Patching-Android-Applications#patching---patching-an-apk) with the [frida-gadget](https://www.frida.re/docs/gadget/). Objection communicates then using a Python API with the mobile phone through the installed Gadget application.